python-checks/src/main/java/org/sonar/python/checks/hotspots/CsrfDisabledCheck.java

```diff
@@ -0,0 +1,326 @@
+    CallExpression callExpr = (CallExpression) subscriptionContext.syntaxNode();
+    Optional.ofNullable(callExpr.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
+      .filter(DANGEROUS_DECORATORS::contains)
+      .ifPresent(fqn -> subscriptionContext.addIssue(callExpr.callee().lastToken(), DISABLING_CSRF_MESSAGE));
+  }
+
+  /** Checks that <code>'WTF_CSRF_ENABLED'</code> setting is not switched off. */
+  private static void flaskWtfCsrfEnabledFalseCheck(SubscriptionContext subscriptionContext) {
+    AssignmentStatement asgn = (AssignmentStatement) subscriptionContext.syntaxNode();
+    // Checks that the left hand side is some kind of subscription of `something['WTF_CSRF_ENABLED']`
+    // Does not check what `something` is - overtainting seems extremely unlikely in this case.
+    boolean isWtfCsrfEnabledSubscription = asgn
+      .lhsExpressions()
+      .stream()
+      .flatMap(exprList -> exprList.expressions().stream())
+      .filter(expr -> expr.is(Tree.Kind.SUBSCRIPTION))
+      .flatMap(s -> ((SubscriptionExpression) s).subscripts().expressions().stream())
+      .anyMatch(isStringSatisfying(s -> "WTF_CSRF_ENABLED".equals(s) || "WTF_CSRF_CHECK_DEFAULT".equals(s)));
+    if (isWtfCsrfEnabledSubscription && Expressions.isFalsy(asgn.assignedValue())) {
+      subscriptionContext.addIssue(asgn.assignedValue(), DISABLING_CSRF_MESSAGE);
+    }
+  }
+
+  /**
+   * Detects <code>class Meta</code> withing <code>FlaskForm</code>-subclasses,
+   * with <code>csrf</code> set to <code>False</code>.
+   */
+  private static void metaCheck(SubscriptionContext subscriptionContext) {
+    ClassDef classDef = (ClassDef) subscriptionContext.syntaxNode();
+    boolean isMetaClass = "Meta".equals(classDef.name().name());
+    if (!isMetaClass) {
+      return;
+    }
+
+    boolean isWithinFlaskForm = Optional.ofNullable(TreeUtils.firstAncestorOfKind(classDef, Tree.Kind.CLASSDEF))
+      .map(parentClassDef -> ((ClassDef) parentClassDef).name().symbol())
+      .flatMap(checkedCast(Symbol.Kind.CLASS, ClassSymbol.class))
+      .filter(parentClassSymbol -> parentClassSymbol.canBeOrExtend("flask_wtf.FlaskForm"))
+      .isPresent();
+    if (!isWithinFlaskForm) {
+      return;
+    }
+
+    classDef.body().statements().forEach(stmt -> {
+      if (stmt.is(Tree.Kind.ASSIGNMENT_STMT)) {
+        AssignmentStatement asgn = (AssignmentStatement) stmt;
+        if (isLhsCalled("csrf").test(asgn) && Expressions.isFalsy(asgn.assignedValue())) {
+          subscriptionContext.addIssue(asgn.assignedValue(), DISABLING_CSRF_MESSAGE);
+        }
+      }
+    });
+  }
+
+  /**
+   * If the value has the specified tree-<code>kind</code>, it is cast as an instance of the specified class, and
+   * returned in a non-empty <code>Optional</code>. Otherwise, returns an empty <code>Optional</code>.
+   */
+  private static <A extends Tree, B extends Tree> Optional<B> checkedCast(Tree.Kind kind, Class<B> c, @Nullable A a) {
```

Comment:
Personally I think we shouldn't need those methods.

Why not use something like `.filter(x -> x.is(Kind.Y)).map(Y.class::cast)` instead of defining these methods? It feels more idiomatic.
At any rate if we're going to use those methods, I believe they should be in a util class where other checks will be able to benefit from them. What do you think @andrea-guarino-sonarsource ?

Reply:
Well...  `Optional.ofNullable(x).filter(x -> x.is(Kind.YK)).map(Y.class::cast)` still looked a bit noisy, especially since it occurred over and over again in several places.

If the method definitions themselves are too distracting, I can inline them for now.

I think there is no need extracting them into an Util class, stuff like that is easy to introduce, but hard to get rid of. If anyone else wants to discuss how one might cut down the noise from the checks and casts, maybe we could collect some proposals later a bit later.